pkg/identity/legacy/principal.go

```diff
@@ -0,0 +1,490 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package legacy
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/sigstore/fulcio/pkg/ca/x509ca"
+	"github.com/sigstore/fulcio/pkg/config"
+	"github.com/sigstore/fulcio/pkg/identity"
+	"github.com/sigstore/fulcio/pkg/oauthflow"
+)
+
+type challengeType int
+
+const (
+	EmailValue challengeType = iota
+	SpiffeValue
+	GithubWorkflowValue
+	KubernetesValue
+	URIValue
+	UsernameValue
+)
+
+// All hostnames for subject and issuer OIDC claims must have at least a
+// top-level and second-level domain
+const minimumHostnameLength = 2
+
+type additionalInfo int
+
+// Additional information that can be added as a cert extension.
+const (
+	githubWorkflowTrigger additionalInfo = iota
+	githubWorkflowSha
+	githubWorkflowName
+	githubWorkflowRepository
+	githubWorkflowRef
+)
+
+type legacyPrincipal struct {
```

Comment:
To confirm, `ChallengeResult` should be identical to `legacyPrincipal`, correct?

Reply:
Thats right. Should be:

- `ChallengeResult` -> `legacyPrincipal`, and
- `FulcioConfig` -> `legacyIssuer`